src/Illuminate/Database/Concerns/ManagesTransactions.php

```diff
@@ -155,11 +156,12 @@ public function commit()
     {
         if ($this->transactions == 1) {
             $this->getPdo()->commit();
+        } elseif ($this->transactions == 0) {
+            throw new TransactionException;
```

Comment:
I suggest to add a message here, given more detail why it can't be commited (because none was started)